packages/browser/src/sdk.ts

```diff
@@ -74,9 +74,16 @@ export const defaultIntegrations = [
  * @see {@link BrowserOptions} for documentation on configuration options.
  */
 export function init(options: BrowserOptions = {}): void {
-  if (options.defaultIntegrations === undefined) {
-    options.defaultIntegrations = defaultIntegrations;
+  options._internal = options._internal || {};
+
+  // Both `defaultIntegrations` and `discoverIntegrations` should be `boolean`, but we are stuck with this type
+  // for backwards compatibility at the momement.
+  if (options.defaultIntegrations !== false) {
+    options._internal.defaultIntegrations = Array.isArray(options.defaultIntegrations)
+      ? options.defaultIntegrations
+      : defaultIntegrations;
```

Comment:
`defaultIntegrations` in `Options`, if not `false`, is of type `Integration[]`: https://github.com/getsentry/sentry-javascript/blob/10e6720e292b7a60a66c1af24ccf80e10f2947a1/packages/types/src/options.ts#L34. Why, after checking if `false`, check again if it's an `Array`?

Reply:
Because we messed up types long time ago, and some people are explicitly doing `defaultIntegrations: true`, which would fail without this check.